core/src/operations/mod.rs

```diff
@@ -363,10 +363,30 @@ pub enum Operation {
     Pipe,
 
     // ----- cryptographic operations -------------------------------------------------------------
-    /// Applies a permutation of Rescue Prime Optimized to the top 12 elements of the stack. The
-    /// rate part of the sponge is assumed to be on top of the stack, and the capacity is expected
-    /// to be deepest in the stack, starting at stack[8]. For an RPO permutation of [A, B, C] where
-    /// A is the capacity, the stack should look like [C, B, A, ...] from the top.
+    /// Applies a single permutation of Rescue Prime Optimized to the top 12 elements of the stack.
+    ///
+    /// # Usage
+    ///
+    /// This operation performs a single permutation of the Rescue Prime Optimized hash function.
+    /// The top 12 elements of the stack are used as the state for the hash function. The stack
+    /// itself is in reverse order, so the first 8 elements in the range 0..8 contains the two
+    /// elements to be absorbed by the hasher, and the last 4 elements in the range 8..12 contains
+    /// the capacity word of the hash function. The result of the permutation is in the middle
+    /// word, in the range 4..8.
```

Comment:
I think it's confusing to say that the stack is in reverse order. The "first" elements of anything are always deepest in the stack, and we do this consistently. I think this phrasing might lead to confusion about whether we mean that the "first" elements are at the top of the stack instead of in the deepest position.

I think what we had before was clearer. Maybe we could clarify it further by specifying that B and C make up the rate and B is the digest (output) of the hash, but I'm not sure that these ranges help, since it becomes a bit unclear whether we're talking about the range of the stack or the range of the hasher state.

Reply:
- remove reverse order, using shallow/deep instead
- using C,B,A instead of A,B,C